src/posthog-core.ts

```diff
@@ -863,6 +863,13 @@ export class PostHog {
             return
         }
 
+        if (properties?.$current_url && !isString(properties?.$current_url)) {
+            logger.error(
+                'Invalid `$current_url` property provided to `posthog.capture`. Input must be a string. Ignoring provided value.'
+            )
+            delete properties?.$current_url
```

Comment:
most annoying feature of GitHub is you can't thread conversations on the main page of the PR @haacked 

> Could we just detect this situation and unpack it for them? After all, they were trying to be helpful. 😆

We sort of will since we always add the correct URL anyway
(and in the replay UI we try and grab the right href if you send `{ href: 'blah' }` instead of 'blah')

but the most recent case I saw we just got `{}` as the URL for some reason, so i thought rather than try and detect  what value they were sending we'll just delete it and set the "correct" value

Reply:
(obvs happy to be cleverer if i'm missing something!)

Reply:
Sounds good to me!

Reply:
<img width="433" alt="Screenshot 2025-03-19 at 19 54 24" src="https://github.com/user-attachments/assets/8ee8fcfc-5fb0-4dd3-a490-05a5fedb012c" />

we get a relatively wide range of values if we query

```
select mat_$current_url
from events
where timestamp <= now()
and timestamp >= now() - interval 7 day
and event = '$pageview'
and mat_$current_url like '{%'
```

Reply:
Oh wow!